### PR TITLE
Region based tuner fixes to extend function, intersection calculation and adding unit tests

### PR DESCRIPTION
**Pull request by arunkarthik-akkart**
This PR addresses the below issues we have on the current tuner :
1. Fix the extend function to the x-max (TUNER_MAX_SIZE) and y-max(TUNER_MAX_RANKS) and adds the unit tests to check for the bounds.
2. Fix the precision on the intersection point from double to long double, which makes the classification of points on the edge with better accuracy.
3. It add unit tests to check if a point is classified to be inside a polygon, outside a polygon or on the edge.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/tuner/nccl_ofi_regions.c [hunk 1]
```diff
@@ -49,7 +49,7 @@ static inline double vcross(nccl_ofi_tuner_point_t a, nccl_ofi_tuner_point_t b)
 }
 
 /* Returns a + s * b */
-static inline nccl_ofi_tuner_point_t vmadd(nccl_ofi_tuner_point_t a, double s, nccl_ofi_tuner_point_t b)
+static inline nccl_ofi_tuner_point_t vmadd(nccl_ofi_tuner_point_t a, long double s, nccl_ofi_tuner_point_t b)
```
Comment (aws-nslick) [inline src/tuner/nccl_ofi_regions.c:45]: nice find.

**Review by bwbarrett — APPROVED** [review]

**Review by rauteric — APPROVED** [review]
